contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -167,6 +176,9 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
             revert MethodNotAllowed(ERR_VALIDATOR_NOT_JOINED);
         }
 
+        uint256 collateral = LibStaking.totalValidatorCollateral(msg.sender);
+        LibSubnetActor.gateValidatorPowerDelta(msg.sender, collateral, collateral + msg.value);
```

Comment:
I think the idea works, but just from gas concern + responsitbility, I'm not sure if SubnetActor should enforce this. 

I can achieve the same check with my own Gater implementation because I know the old and new value. In my custom gater implementation, I can hardcode a total power limit. Then with each validator update, I can calculate the new total power with `oldTotalPower = oldTotalPower + validatorNewPower - validatorOldPower`. Then I can do the comparison with the hardcoded limit. 

I feel (I could be wrong) the current interface is general enough for first iteration, adding total network power might incur a gas cost to users who dont need it. And I can achieve that with my custom implementation.